**Linux** [Build 9207](http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=87854&buildTypeId=WFPR) outcome was **FAILURE** using a merge of 24e1f42c861267b553e4e199a5ee0d498b1ab3b2
Summary: Tests failed: 2 (2 new), passed: 3309, ignored: 191 Build time: 00:59:07
### Failed tests
```
org.jboss.as.test.integration.messaging.jms.definitions.JMSResourceManagementTestCase: org.jboss.as.test.integration.messaging.jms.definitions.JMSResourceDefinitionsTestCase: 	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:83)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:64)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deploy(ArchiveDeployer.java:46)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.deploy(CommonDeployableContainer.java:144)
Caused by: java.lang.Exception: {
    "WFLYCTL0080: Failed services" => {"jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".INSTALL" => "org.jboss.msc.service.StartException in service jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".INSTALL: WFLYSRV0153: Failed to process phase INSTALL of deployment \"JMSResourceDefinitionsTestCase.jar\"
    Caused by: java.lang.NullPointerException"},
    "WFLYCTL0180: Services with missing/unavailable dependencies" => [
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.ORB is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]",
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.InstanceName is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]",
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.InAppClientContainer is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]",
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.HandleDelegate is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]",
        "jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".batch.environment is missing [jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".beanmanager]",
        "jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".weld.weldClassIntrospector is missing [jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".beanmanager]",
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.Validator is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]",
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.ValidatorFactory is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]"
    ]
}
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.getActionResult(ServerDeploymentPlanResultFuture.java:134)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.getResultFromNode(ServerDeploymentPlanResultFuture.java:123)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.get(ServerDeploymentPlanResultFuture.java:85)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.get(ServerDeploymentPlanResultFuture.java:42)
	at org.jboss.as.controller.client.helpers.standalone.ServerDeploymentHelper.deploy(ServerDeploymentHelper.java:55)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:77)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:64)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deploy(ArchiveDeployer.java:46)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.deploy(CommonDeployableContainer.java:144)
------- Stdout: -------
 [0m11:12:34,236 INFO  [org.jboss.weld.deployer] (MSC service thread 1-3) WFLYWELD0010: Stopping weld service for deployment TransactionScopedJMSContextTestCase.jar
 [0m [0m11:12:34,237 INFO  [org.wildfly.extension.messaging-activemq] (MSC service thread 1-3) WFLYMSGAMQ0006: Unbound messaging object to jndi name java:app/TransactionScopedJMSContextTestCase
 [0m [0m11:12:34,238 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-1) WFLYSRV0028: Stopped deployment TransactionScopedJMSContextTestCase.jar (runtime-name: TransactionScopedJMSContextTestCase.jar) in 7ms
 [0m [0m11:12:34,240 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 72) WFLYCLINF0003: Stopped client-mappings cache from ejb container
 [0m [0m11:12:34,245 INFO  [org.jboss.as.repository] (management-handler-thread - 3) WFLYDR0002: Content removed from location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/38/4d329cb1ba0dda9bd5856baec1d3fd2a91e25a/content
 [0m [0m11:12:34,245 INFO  [org.jboss.as.server] (management-handler-thread - 3) WFLYSRV0009: Undeployed "TransactionScopedJMSContextTestCase.jar" (runtime-name: "TransactionScopedJMSContextTestCase.jar")
 [0marchive = JMSResourceDefinitionsTestCase.jar:
/org/
/org/jboss/
/org/jboss/as/
/org/jboss/as/test/
/org/jboss/as/test/integration/
/org/jboss/as/test/integration/messaging/
/org/jboss/as/test/integration/messaging/jms/
/org/jboss/as/test/integration/messaging/jms/definitions/
/org/jboss/as/test/integration/messaging/jms/definitions/JMSResourceDefinitionsTestCase$StoreVaultedPropertyTask.class
/org/jboss/as/test/integration/messaging/jms/definitions/JMSResourceManagementTestCase.class
/org/jboss/as/test/integration/messaging/jms/definitions/JMSResourceDefinitionsTestCase.class
/org/jboss/as/test/integration/messaging/jms/definitions/MessagingBean.class
/META-INF/
/META-INF/jboss-permissions.xml
/META-INF/ejb-jar.xml
/META-INF/beans.xml


org.jboss.as.test.integration.messaging.jms.definitions.JMSResourceManagementTestCase: 	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:83)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:64)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deploy(ArchiveDeployer.java:46)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.deploy(CommonDeployableContainer.java:144)
Caused by: java.lang.Exception: {
    "WFLYCTL0080: Failed services" => {"jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".INSTALL" => "org.jboss.msc.service.StartException in service jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".INSTALL: WFLYSRV0153: Failed to process phase INSTALL of deployment \"JMSResourceDefinitionsTestCase.jar\"
    Caused by: java.lang.NullPointerException"},
    "WFLYCTL0180: Services with missing/unavailable dependencies" => [
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.InstanceName is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]",
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.Validator is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]",
        "jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".batch.environment is missing [jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".beanmanager]",
        "jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".weld.weldClassIntrospector is missing [jboss.deployment.unit.\"JMSResourceDefinitionsTestCase.jar\".beanmanager]",
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.InAppClientContainer is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]",
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.ORB is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]",
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.HandleDelegate is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]",
        "jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean.ValidatorFactory is missing [jboss.naming.context.java.comp.JMSResourceDefinitionsTestCase.JMSResourceDefinitionsTestCase.MessagingBean]"
    ]
}
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.getActionResult(ServerDeploymentPlanResultFuture.java:134)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.getResultFromNode(ServerDeploymentPlanResultFuture.java:123)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.get(ServerDeploymentPlanResultFuture.java:85)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.get(ServerDeploymentPlanResultFuture.java:42)
	at org.jboss.as.controller.client.helpers.standalone.ServerDeploymentHelper.deploy(ServerDeploymentHelper.java:55)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:77)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:64)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deploy(ArchiveDeployer.java:46)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.deploy(CommonDeployableContainer.java:144)
------- Stdout: -------
 [0m11:12:34,424 INFO  [org.jboss.as.repository] (management-handler-thread - 1) WFLYDR0001: Content added at location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/a9/87c1055f2c62d960d429ab281809a5a21091b0/content
 [0m [0m11:12:34,426 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-3) WFLYSRV0027: Starting deployment of "JMSResourceDefinitionsTestCase.jar" (runtime-name: "JMSResourceDefinitionsTestCase.jar")
 [0m [33m11:12:34,432 WARN  [org.jboss.as.dependency.private] (MSC service thread 1-3) WFLYSRV0018: Deployment "deployment.JMSResourceDefinitionsTestCase.jar" is using a private module ("org.wildfly.security.manager:main") which may be changed or removed in future versions without notice.
 [0m [0m11:12:34,435 INFO  [org.jboss.weld.deployer] (MSC service thread 1-3) WFLYWELD0003: Processing weld deployment JMSResourceDefinitionsTestCase.jar
 [0m [0m11:12:34,436 INFO  [org.jboss.as.ejb3.deployment] (MSC service thread 1-3) WFLYEJB0473: JNDI bindings for session bean named 'MessagingBean' in deployment unit 'deployment "JMSResourceDefinitionsTestCase.jar"' are as follows:

	java:global/JMSResourceDefinitionsTestCase/MessagingBean!org.jboss.as.test.integration.messaging.jms.definitions.MessagingBean
	java:app/JMSResourceDefinitionsTestCase/MessagingBean!org.jboss.as.test.integration.messaging.jms.definitions.MessagingBean
	java:module/MessagingBean!org.jboss.as.test.integration.messaging.jms.definitions.MessagingBean
	java:global/JMSResourceDefinitionsTestCase/MessagingBean
	java:app/JMSResourceDefinitionsTestCase/MessagingBean
	java:module/MessagingBean

 [0m [31m11:12:34,440 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-3) MSC000001: Failed to start service jboss.deployment.unit."JMSResourceDefinitionsTestCase.jar".INSTALL: org.jboss.msc.service.StartException in service jboss.deployment.unit."JMSResourceDefinitionsTestCase.jar".INSTALL: WFLYSRV0153: Failed to process phase INSTALL of deployment "JMSResourceDefinitionsTestCase.jar"
	at org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:154)
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1948)
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.run(ServiceControllerImpl.java:1881)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
	at org.wildfly.extension.messaging.activemq.deployment.JMSDestinationDefinitionInjectionSource.getResourceValue(JMSDestinationDefinitionInjectionSource.java:126)
	at org.jboss.as.ee.component.deployers.ModuleJndiBindingProcessor.addJndiBinding(ModuleJndiBindingProcessor.java:211)


```

--- a/connector/src/main/java/org/jboss/as/connector/deployers/ra/AdministeredObjectDefinitionInjectionSource.java
+++ b/connector/src/main/java/org/jboss/as/connector/deployers/ra/AdministeredObjectDefinitionInjectionSource.java
@@ -68,6 +68,10 @@ public class AdministeredObjectDefinitionInjectionSource extends ResourceDefinit
         this.resourceAdapter = resourceAdapter;
     }
 
+    public void addProperty(String key, String value) {
+        properties.put(key, value);
+    }
+
     public void getResourceValue(final ResolutionContext context, final ServiceBuilder<?> serviceBuilder, final DeploymentPhaseContext phaseContext, final Injector<ManagedReferenceFactory> injector) throws DeploymentUnitProcessingException {
         final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
         final Module module = deploymentUnit.getAttachment(org.jboss.as.server.deployment.Attachments.MODULE);

--- a/connector/src/main/java/org/jboss/as/connector/deployers/ra/ConnectionFactoryDefinitionInjectionSource.java
+++ b/connector/src/main/java/org/jboss/as/connector/deployers/ra/ConnectionFactoryDefinitionInjectionSource.java
@@ -74,6 +74,10 @@ public class ConnectionFactoryDefinitionInjectionSource extends ResourceDefiniti
         this.resourceAdapter = resourceAdapter;
     }
 
+    public void addProperty(String key, String value) {
+        properties.put(key, value);
+    }
+
     public void getResourceValue(final ResolutionContext context, final ServiceBuilder<?> serviceBuilder, final DeploymentPhaseContext phaseContext, final Injector<ManagedReferenceFactory> injector) throws DeploymentUnitProcessingException {
         final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
         final Module module = deploymentUnit.getAttachment(org.jboss.as.server.deployment.Attachments.MODULE);

--- a/feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/messaging-activemq/main/module.xml
+++ b/feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/messaging-activemq/main/module.xml
@@ -39,6 +39,7 @@
         <module name="javax.api"/>
         <module name="javax.enterprise.api"/>
         <module name="javax.jms.api"/>
+        <module name="javax.resource.api" />
         <module name="javax.transaction.api"/>
         <module name="org.apache.activemq.artemis"/>
         <module name="org.apache.activemq.artemis.ra"/>

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/deployment/JMSConnectionFactoryDefinitionInjectionSource.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/deployment/JMSConnectionFactoryDefinitionInjectionSource.java
@@ -42,6 +42,9 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
+import javax.resource.spi.TransactionSupport;
+
+import org.jboss.as.connector.deployers.ra.ConnectionFactoryDefinitionInjectionSource;
 import org.jboss.as.connector.services.resourceadapters.ConnectionFactoryReferenceFactoryService;
 import org.jboss.as.controller.OperationFailedException;
 import org.jboss.as.controller.PathAddress;
@@ -148,10 +151,33 @@ public class JMSConnectionFactoryDefinitionInjectionSource extends ResourceDefin
     public void getResourceValue(ResolutionContext context, ServiceBuilder<?> serviceBuilder, DeploymentPhaseContext phaseContext, Injector<ManagedReferenceFactory> injector) throws DeploymentUnitProcessingException {
         final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
 
-        try {
-            startedPooledConnectionFactory(context, jndiName, serviceBuilder, phaseContext.getServiceTarget(), deploymentUnit, injector);
-        } catch (OperationFailedException e) {
-            throw new DeploymentUnitProcessingException(e);
+        // if the resourceAdapter is empty, we use the embedded Artemis server to create a pooled-connection-factory
+        if (resourceAdapter == null || resourceAdapter.isEmpty()) {
+            try {
+                startedPooledConnectionFactory(context, jndiName, serviceBuilder, phaseContext.getServiceTarget(), deploymentUnit, injector);
+            } catch (OperationFailedException e) {
+                throw new DeploymentUnitProcessingException(e);
+            }
+        } else {
+            // delegate to the resource-adapter subsystem to create a generic JCA connection factory.
+            ConnectionFactoryDefinitionInjectionSource cfdis = new ConnectionFactoryDefinitionInjectionSource(jndiName, interfaceName, resourceAdapter);
+            cfdis.setMaxPoolSize(maxPoolSize);
+            cfdis.setMinPoolSize(minPoolSize);
+            cfdis.setTransactionSupportLevel(transactional ? TransactionSupport.TransactionSupportLevel.XATransaction : TransactionSupport.TransactionSupportLevel.NoTransaction);
+            // transfer all the generic properties + the additional properties specific to the JMSConnectionFactoryDefinition
+            for (Map.Entry<String, String> property : properties.entrySet()) {
+                cfdis.addProperty(property.getKey(), property.getValue());
+            }
+            if (!user.isEmpty()) {
+                cfdis.addProperty("user", user);
+            }
+            if (!password.isEmpty()) {
+                cfdis.addProperty("password", password);
+            }
+            if (!clientId.isEmpty()) {
+                cfdis.addProperty("clientId", clientId);
+            }
+            cfdis.getResourceValue(context, serviceBuilder, phaseContext, injector);
         }
     }
 

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/deployment/JMSDestinationDefinitionInjectionSource.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/deployment/JMSDestinationDefinitionInjectionSource.java
@@ -32,25 +32,22 @@ import static org.wildfly.extension.messaging.activemq.CommonAttributes.SELECTOR
 import static org.wildfly.extension.messaging.activemq.CommonAttributes.SERVER;
 import static org.wildfly.extension.messaging.activemq.logging.MessagingLogger.ROOT_LOGGER;
 
+import java.util.Map;
+
 import javax.jms.Destination;
 import javax.jms.Queue;
 import javax.jms.Topic;
 
+import org.jboss.as.connector.deployers.ra.AdministeredObjectDefinitionInjectionSource;
 import org.jboss.as.controller.PathAddress;
 import org.jboss.as.controller.PathElement;
 import org.jboss.as.ee.component.InjectionSource;
 import org.jboss.as.ee.resource.definition.ResourceDefinitionInjectionSource;
-import org.jboss.as.server.deployment.Attachments;
-import org.jboss.as.server.deployment.DeploymentResourceSupport;
-import org.wildfly.extension.messaging.activemq.MessagingExtension;
-import org.wildfly.extension.messaging.activemq.MessagingServices;
-import org.wildfly.extension.messaging.activemq.jms.JMSQueueConfigurationRuntimeHandler;
-import org.wildfly.extension.messaging.activemq.jms.JMSQueueService;
-import org.wildfly.extension.messaging.activemq.jms.JMSTopicConfigurationRuntimeHandler;
-import org.wildfly.extension.messaging.activemq.jms.JMSTopicService;
 import org.jboss.as.naming.ContextListAndJndiViewManagedReferenceFactory;
 import org.jboss.as.naming.ManagedReferenceFactory;
+import org.jboss.as.server.deployment.Attachments;
 import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentResourceSupport;
 import org.jboss.as.server.deployment.DeploymentUnit;
 import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
 import org.jboss.dmr.ModelNode;
@@ -61,6 +58,12 @@ import org.jboss.msc.service.ServiceBuilder;
 import org.jboss.msc.service.ServiceController;
 import org.jboss.msc.service.ServiceName;
 import org.jboss.msc.service.ServiceTarget;
+import org.wildfly.extension.messaging.activemq.MessagingExtension;
+import org.wildfly.extension.messaging.activemq.MessagingServices;
+import org.wildfly.extension.messaging.activemq.jms.JMSQueueConfigurationRuntimeHandler;
+import org.wildfly.extension.messaging.activemq.jms.JMSQueueService;
+import org.wildfly.extension.messaging.activemq.jms.JMSTopicConfigurationRuntimeHandler;
+import org.wildfly.extension.messaging.activemq.jms.JMSTopicService;
 import org.wildfly.extension.messaging.activemq.jms.WildFlyBindingRegistry;
 
 /**
@@ -119,6 +122,24 @@ public class JMSDestinationDefinitionInjectionSource extends ResourceDefinitionI
     }
 
     public void getResourceValue(final InjectionSource.ResolutionContext context, final ServiceBuilder<?> serviceBuilder, final DeploymentPhaseContext phaseContext, final Injector<ManagedReferenceFactory> injector) throws DeploymentUnitProcessingException {
+        // if the resourceAdapter is empty, we use the embedded Artemis server to create the JMS destination
+        if (resourceAdapter == null || resourceAdapter.isEmpty()) {
+            startActiveMQDestination(context, serviceBuilder, phaseContext, injector);
+        } else {
+            // delegate to the resource-adapter subsystem to create a generic JCA admin object.
+            AdministeredObjectDefinitionInjectionSource aodis = new AdministeredObjectDefinitionInjectionSource(jndiName, className, resourceAdapter);
+            aodis.setInterface(interfaceName);
+            aodis.setDescription(description);
+            // transfer all the generic properties
+            for (Map.Entry<String, String> property : properties.entrySet()) {
+                aodis.addProperty(property.getKey(), property.getValue());
+            }
+            aodis.getResourceValue(context, serviceBuilder, phaseContext, injector);
+
+        }
+    }
+
+    private void startActiveMQDestination(ResolutionContext context, ServiceBuilder<?> serviceBuilder, DeploymentPhaseContext phaseContext, Injector<ManagedReferenceFactory> injector) throws DeploymentUnitProcessingException {
         final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
         final String uniqueName = uniqueName(context);
         try {

